### PR TITLE
infosync: improve error log for http client

### LINKED ISSUE #33900: improve error log when to load placement rule from pd.

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #33900

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/infosync/info.go [hunk 1]
```diff
@@ -405,6 +406,8 @@ func doRequest(ctx context.Context, addrs []string, route, method string, body i
 			}
 			terror.Log(res.Body.Close())
 			return bodyBytes, err
+		} else {
+			logutil.BgLogger().Error("fail to doRequest", zap.Error(err), zap.String("server addr", addr))
```
Comment (lcwangchao) [inline domain/infosync/info.go:410]: 1. level Warn is better? 
2. message to "fail to doRequest, retry next address" 
3. I think we can add more information like addressIndex, method and route.

Comment (hawkingrei) [reply domain/infosync/info.go]: Done

File: domain/infosync/info.go [hunk 2]
```diff
@@ -406,6 +413,14 @@ func doRequest(ctx context.Context, addrs []string, route, method string, body i
 			terror.Log(res.Body.Close())
 			return bodyBytes, err
 		}
+		logutil.BgLogger().Warn("fail to doRequest, retry next address",
+			zap.Error(err),
+			zap.String("method", method),
+			zap.String("hosts", addr),
+			zap.String("url", url),
+			zap.Int("http status", res.StatusCode),
```
Comment (lcwangchao) [inline domain/infosync/info.go:421]:
```suggestion
			zap.Int("http status", res.StatusCode),
```

I think the res will be nil here

Comment (lcwangchao) [inline domain/infosync/info.go:421]:
```suggestion
```

**Review by lcwangchao — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**